subsys/bluetooth/shell/audio.c

```diff
@@ -822,18 +822,97 @@ static int cmd_qos(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+static uint16_t strmeta(const char *name)
+{
+	if (strcmp(name, "Unspecified") == 0) {
+		return BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED;
+	} else if (strcmp(name, "Conversational") == 0) {
+		return BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL;
+	} else if (strcmp(name, "Media") == 0) {
+		return BT_AUDIO_CONTEXT_TYPE_MEDIA;
+	} else if (strcmp(name, "Game") == 0) {
+		return BT_AUDIO_CONTEXT_TYPE_GAME;
+	} else if (strcmp(name, "Instructional") == 0) {
+		return BT_AUDIO_CONTEXT_TYPE_INSTRUCTIONAL;
+	} else if (strcmp(name, "VoiceAssistants") == 0) {
+		return BT_AUDIO_CONTEXT_TYPE_VOICE_ASSISTANTS;
+	} else if (strcmp(name, "Live") == 0) {
+		return BT_AUDIO_CONTEXT_TYPE_LIVE;
+	} else if (strcmp(name, "SoundEffects") == 0) {
+		return BT_AUDIO_CONTEXT_TYPE_SOUND_EFFECTS;
+	} else if (strcmp(name, "Notifications") == 0) {
+		return BT_AUDIO_CONTEXT_TYPE_NOTIFICATIONS;
+	} else if (strcmp(name, "Ringtone") == 0) {
+		return BT_AUDIO_CONTEXT_TYPE_RINGTONE;
+	} else if (strcmp(name, "Alerts") == 0) {
+		return BT_AUDIO_CONTEXT_TYPE_ALERTS;
+	} else if (strcmp(name, "EmergencyAlarm") == 0) {
+		return BT_AUDIO_CONTEXT_TYPE_EMERGENCY_ALARM;
+	}
+
+	return 0u;
+}
+
+static int handle_metadata_update(const char *meta_str,
+				  struct bt_codec_data *meta_out[],
+				  size_t *meta_count_out)
+{
+	static struct bt_codec_data meta[CONFIG_BT_CODEC_MAX_METADATA_COUNT];
+	size_t meta_count;
+
+	/* We create a copy of the preset meta, as the presets cannot be modified */
+	meta_count = default_preset->preset.codec.meta_count;
+	(void)memset(meta, 0, sizeof(meta));
+	for (size_t i = 0U; i < meta_count; i++) {
+		(void)memcpy(meta[i].value,
+			     default_preset->preset.codec.meta[i].data.data,
+			     default_preset->preset.codec.meta[i].data.data_len);
+		meta[i].data.data_len = default_preset->preset.codec.meta[i].data.data_len;
+		meta[i].data.data = meta[i].value;
+	}
+
+	if (meta_str != NULL) {
+		uint16_t context;
+
+		context = strmeta(meta_str);
+		if (context == 0) {
+			return -ENOEXEC;
+		}
+
+		/* TODO: Check the type nad only overwrite the streaming context */
+		sys_put_le16(context, meta[0].value);
+	}
+
+	*meta_count_out = meta_count;
+	*meta_out = meta;
+
+	return 0;
+}
+
 static int cmd_enable(const struct shell *sh, size_t argc, char *argv[])
 {
+	struct bt_codec_data *meta;
+	size_t meta_count;
 	int err;
 
 	if (default_stream == NULL) {
 		shell_error(sh, "Not connected");
 		return -ENOEXEC;
 	}
 
-	err = bt_audio_stream_enable(default_stream,
-				     default_preset->preset.codec.meta,
-				     default_preset->preset.codec.meta_count);
+	if (argc > 1) {
+		err = handle_metadata_update(argv[1], &meta, &meta_count);
+
+		if (err != 0) {
+			shell_error(sh, "Unable to handle metadata update: %d",
+				    err);
+			return err;
+		}
+	} else {
+		err = handle_metadata_update(NULL, &meta, &meta_count);
```

Comment:
erer is never used, isn't it then better to write
```suggestion
		(void)handle_metadata_update(NULL, &meta, &meta_count);
```

Reply:
Moved the error check to after this one, so that it is handled.